windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -490,33 +490,35 @@ UwbConnector::HandleNotifications(std::stop_token stopToken)
             uwbNotificationDataBuffer.resize(bytesRequired);
             PLOG_DEBUG << "IOCTL_UWB_NOTIFICATION attempt #" << (i + 1) << " with " << std::size(uwbNotificationDataBuffer) << "-byte buffer";
             BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_NOTIFICATION, nullptr, 0, std::data(uwbNotificationDataBuffer), std::size(uwbNotificationDataBuffer), &bytesRequired, &m_notificationOverlapped);
+            DWORD lastError = GetLastError();
+            HRESULT hr = HRESULT_FROM_WIN32(lastError);
+            PLOG_DEBUG << "IOCTL_UWB_NOTIFICATION attempt #" << (i + 1) << " with " << std::size(uwbNotificationDataBuffer) << "-byte buffer completed " << bytesRequired << " bytes required hr=" << std::showbase << std::hex << hr << " lastError " << std::showbase << std::hex << lastError;
             if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
-                DWORD lastError = GetLastError();
+                lastError = GetLastError();
                 if (lastError == ERROR_IO_PENDING) {
-                    // I/O has been pended, wait for it synchronously
+                    // I/O has been pended, wait for it synchronously, but in an interruptable manner.
                     if (!LOG_IF_WIN32_BOOL_FALSE(GetOverlappedResult(handleDriver.get(), &m_notificationOverlapped, &bytesRequired, TRUE /* wait */))) {
                         lastError = GetLastError();
-                        HRESULT hr = HRESULT_FROM_WIN32(lastError);
-                        switch (lastError) {
-                        case ERROR_INSUFFICIENT_BUFFER: // occurs for overlapped i/o
-                        case ERROR_MORE_DATA:           // occurs for synchronous i/o
-                            LOG_DEBUG << "IOCTL_UWB_NOTIFICATION completed, hr=" << std::showbase << std::hex << hr;
-                            // driver indicated buffer was too small and put required size in 'bytesRequired'. Retry with new size.
+                        hr = HRESULT_FROM_WIN32(lastError);
+                        if (lastError == ERROR_INSUFFICIENT_BUFFER || lastError == ERROR_MORE_DATA) {
+                            // Driver indicated buffer was too small and put required size in 'bytesRequired'. Retry with new size.

```

Comment:
This flow may need to be updated now that we have new information from Wei regarding the DeviceIoControl. The required size should not be put in bytesRequired anymore - it should be set in the "size" field of the structs. This change has been made in the CX but should also be made in the Simulator driver I'm assuming

Reply:
Yes, it'll have to be updated, but that will be done in another PR.